hyperapp.d.ts

```diff
@@ -96,7 +96,7 @@ export function app<State, Actions>(
 
 declare global {
   namespace JSX {
-    interface Element<Data> extends VNode<object> {}
+    interface Element extends VNode<any> {}
```

Comment:
This change came from [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f2982c175477516d52bdc40191d7829d0531ba35/types/react/index.d.ts#L3742).

Reply:
could you explain me why you need to add the state and actions object into the generics for a `components` and `h`, is there a reason for that? what doesn't work without that currently?

Reply:
It is because `View` type is generics:

https://github.com/hyperapp/hyperapp/blob/9031077fcb9bf760fd4e86593fda284ce000bb3b/hyperapp.d.ts#L75-L77

This PR aims to allow `Component` type to return `View` so I had to add `State` and `Actions` to `Component`.

Reply:
@dancespiele See https://github.com/hyperapp/hyperapp/releases/tag/1.2.0

Component are not able to return a `View`. If you are not using the Lazy Components feature or you prefer to type the `View` arguments by your self it is also possible.